bundles/org.openhab.binding.fsinternetradio/src/main/java/org/openhab/binding/fsinternetradio/internal/FSInternetRadioDiscoveryParticipant.java

```diff
@@ -146,72 +148,66 @@
     }
 
     @Override
+    @Nullable
     public DiscoveryResult createResult(RemoteDevice device) {
-        final ThingUID uid = getThingUID(device);
+        ThingUID uid = getThingUID(device);
         if (uid != null) {
-            final Map<String, Object> properties = new HashMap<>(1);
-            final String ip = getIp(device);
-            if (ip != null) {
-                properties.put(CONFIG_PROPERTY_IP, ip);
+            Map<String, Object> properties = new HashMap<>(1);
+            Optional<String> ip = getIp(device);
+            if (ip.isPresent()) {
+                properties.put(CONFIG_PROPERTY_IP, ip.get());
 
                 // add manufacturer and model, if provided
-                final String manufacturer = getManufacturer(device);
-                if (manufacturer != null) {
+                Optional<String> manufacturer = getManufacturer(device);
+                Optional<String> model = getModel(device);
+                if (manufacturer.isPresent()) {
                     properties.put(PROPERTY_MANUFACTURER, manufacturer);
                 }
-                final String model = getModel(device) != null ? getModel(device) : getFriendlyName(device);
-                if (model != null) {
-                    properties.put(PROPERTY_MODEL, model);
-                }
-                final String thingName = (manufacturer == null) && (getModel(device) == null) ? getFriendlyName(device)
-                        : device.getDisplayString();
-                return DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(thingName).build();
+                properties.put(PROPERTY_MODEL, model.isPresent() ? model : getFriendlyName(device));
+                Optional<String> thingName = !manufacturer.isPresent() && !model.isPresent() ? getFriendlyName(device)
```

Comment:
No need for an `Optional here`, you get the value in the next line anyway.

Reply:
see above